modules/directory/init.zsh

```diff
@@ -28,3 +28,19 @@ unsetopt CLOBBER            # Do not overwrite existing files with > and >>.
 
 alias d='dirs -v'
 for index ({1..9}) alias "$index"="cd +${index}"; unset index
+
+if (is-at-least 4.1) \
+    && zstyle -t ':prezto:module:directory' persist-dirstack
+then
+    if [[ -f ~/.zdirs ]] && [[ ${#dirstack[*]} -eq 0 ]]; then
+        echo "\n" >> ~/.zdirs
```

Comment:
This shouldn't go here. I'm fairly certain it will cause a newline to be appended each time a new shell is opened (without changing directories). Making sure there's a newline should probably go into the `_persist_dirstack` function.